test/cpp/jit/backend_with_compiler.cpp

```diff
@@ -0,0 +1,159 @@
+  virtual ~BackendWithCompiler() = default;
+
+  // Since the actual compilation is done AOT,
+  c10::impl::GenericDict compile(
+      c10::IValue processed,
+      c10::impl::GenericDict method_compile_spec) override {
+    //    return processed.toGenericDict();
+    auto dict = processed.toGenericDict();
+    auto handles = c10::Dict<std::string, std::vector<std::string>>();
+    for (const auto& kv : dict) {
+      auto tokens = parseMethodHandle(kv.value().toStringRef());
+      handles.insert(kv.key().toStringRef(), tokens);
+    }
+    return c10::impl::toGenericDict(handles);
+  }
+
+  c10::impl::GenericList execute(
+      c10::IValue handle,
+      c10::impl::GenericList inputs) override {
+    //    TORCH_INTERNAL_ASSERT(handle.isString());
+    TORCH_INTERNAL_ASSERT(inputs.size() == 2);
+    c10::IValue val0 = inputs[0];
+    at::Tensor x = val0.toTensor();
+    c10::IValue val1 = inputs[1];
+    at::Tensor h = val1.toTensor();
+
+    c10::List<at::Tensor> output_list;
+    double scalar_val = 1.0;
+    for (const auto& token : handle.toList()) {
+      IValue val = token;
+      auto instruction = std::string(IValue(token).toStringRef());
+      double const_val = 1.0;
+      if (instruction.rfind("prim::Constant", 0) == 0) {
+        TORCH_CHECK(
+            instruction.size() > 15,
+            "Constant value is expected in ",
+            instruction);
+        auto sub = instruction.substr(15);
+        const_val = stod(sub);
+      } else if (token == "aten::add") {
+        output_list.emplace_back(x.add_(h, const_val));
+      } else if (token == "aten::sub") {
+        output_list.emplace_back(x.sub_(h, const_val));
+      } else {
+        TORCH_CHECK(
+            false,
+            "Instruction, ",
+            instruction,
+            " is not supported. ",
+            "Contact the backend POC for details. ");
+      }
+    }
+    return c10::impl::toList(output_list);
+  }
+};
+
+// For this backend, the actual compilation happens in preprocess functoin AOT.
+// Put here for demonstration of backend
+// as a whole piece. It's used when compilation is required. A dummy function
+// can be passed when there's no usage of compilation in runtime backend lib.
+c10::IValue preprocess(
+    const Module& mod,
+    const c10::Dict<IValue, IValue>& method_compile_spec) {
+  // The output of this process would produce a dictionary
+  // Key: method name.
+  // Val: compiled blob (represented by a string).
+  c10::Dict<IValue, IValue> compiled(StringType::get(), StringType::get());
+  for (const auto& method : mod.get_methods()) {
+    auto graph = method.function().graph()->copy();
+    auto key = method.name();
+    std::stringstream ss;
+    for (const auto& node : graph->nodes()) {
+      switch (node->kind()) {
+        case prim::Constant:
+          ss << node->kind().toDisplayString() << "#"
+             << toIValue(node->output()).value();
+          break;
+        case aten::add:
```

Comment:
By the way aten::add vs aten::add_. The runtime actually executes in place version rather than aten::add.

Reply:
Good to know. Thanks!